lcls_naming_tool/web/app.py

```diff
@@ -1,31 +1,32 @@
-import sys
 import os
-from flask import Flask, render_template, request, url_for, flash, redirect
-from lcls_naming_tool import get_version, load_taxons, validate
+import sys
 
+from flask import Flask, flash, redirect, render_template, request, url_for
+
+from lcls_naming_tool import get_version, load_taxons, validate
 
-sys.path.append(os.path.abspath('../'))
+sys.path.append(os.path.abspath("../"))
 
 app = Flask(__name__)
-app.config['SECRET_KEY'] = 'your secret key'
+app.config["SECRET_KEY"] = "your secret key"
```

Comment:
Does our flask app need this?  what secrets are we dealing with?

Reply:
I deleted this line in the past and it broke the web app. I'll revisit and remove it. Still waiting on IT to give me access to `ssh psca@psctlws01` to test out changes to app.py. I used to login through rhel7 or pslogin but I can't seem to access those servers anymore.

Reply:
> I used to login through rhel7 or pslogin but can't seem to access those servers anymore.

`pslogin` is gone as of Sept 30
You should still have access to `psbuild-rhel7` if on the VPN or via logging in through `s3dflogin`

Reply:
Oh that worked, thanks!

Reply:
Can you get to the server if you jump through s3dflogin?  The old `pslogin` bastion nodes were retired a bit ago

Reply:
Right now, `s3dflogin` doesn't seem to let me ssh to `psca@psctlws01`. I get an error message `ssh: Could not resolve hostname psctlws01: Name or service not known.` I sent IT an email last week and another today but haven't heard from them yet.

Reply:
It's intended that these servers aren't linked.
I asked IT a month ago when they announced `pslogin` was going away: https://jira.slac.stanford.edu/browse/ECS-6209
Only a few ps- servers are accessible from `s3dflogin`.
You have to `ssh` to `s3dflogin`, then to `psbuild-rhel7`, then to where you want to go

OR you have to get on the VPN and then ssh to `psbuild-rhel7`

Reply:
Back on the topic of the secret key, it looks like it's for signing session cookies etc.  The default is `None`, so I'm not really sure why we need this here
https://flask.palletsprojects.com/en/2.3.x/config/#SECRET_KEY

Reply:
For some reason deleting app.config['SECRET_KEY'] = 'your secret key' breaks the site. I tried setting = None and that broke the site too. It seems to be only used for user sessions and not used to protect any persistent information, so I randomly generated the SECRET_KEY with os.urandom(24) and the website works again. The only downside seems to be that it resets cookies but no cookies are set on the site, so it shouldn't affect the naming tool.

https://www.reddit.com/r/flask/comments/m0z7s1/comment/gqb1ko5/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

Reply:
If everything looks okay, can this be approved and merged?

Reply:
I'm still not entirely satisfied with the presence of this key, mostly because we don't understand its affect on the tool.  If we are to be using some secret in this app, we need to be more principled in how we store and access it.  Even if it's randomly generated in this library, we shouldn't be storing that information in the repository.

We need to spend some more time understanding why this is necessary for our app.  Can you make a follow-up issue so we don't lose track of this?

Reply:
If the secret key is actually found to be necessary, we should set its configuration from a separate file https://flask.palletsprojects.com/en/2.0.x/config/#configuring-from-python-files

Reply:
@janeliu-slac  in said issue, could you copy the error/traceback that you get when you remove the secret key?  It may help us figure out what is necessary to change

Reply:
Here's a fun zero upvote stackoverflow discussion about this https://stackoverflow.com/questions/56773559/how-bad-is-it-to-run-flask-without-a-secret-key

Reply:
I did some digging.  The SECRET_KEY is used to sign sessions and encrypt cookies, which allows us to persist data across requests.  We don't store any user / login data, but we do use `flask.flash`, which is short hand for "store the last request made and make it available for only the next request".  This is the command that requires the secret key to operate, and without it we throw a `RuntimeError`.  We use it to display the PV name submitted alongside the valid/invalid message (information from the last PUT is made available in the next GET)

I suppose it's fine to leave this here, but the way we're managing the key is still bad practice

Reply:
Created a new ticket https://jira.slac.stanford.edu/browse/ECS-6563. I checked with Murali about whether SECRET_KEY can be removed. He said "Technically; yes. We do not rely on this for authentication anyways. But if you use cookies or any such thing; no. I'd leave it in there and pick it up [the password for SECRET_KEY] from an environment variable."